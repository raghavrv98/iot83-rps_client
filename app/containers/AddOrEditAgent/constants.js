/*
 *
 * AddOrEditAgent constants
 *
 */

export const CREATE_AGENT_REQUEST = "app/AddOrEditAgent/CREATE_AGENT_REQUEST";
export const CREATE_AGENT_SUCCESS = "app/AddOrEditAgent/CREATE_AGENT_SUCCESS";
export const CREATE_AGENT_FAILURE = "app/AddOrEditAgent/CREATE_AGENT_FAILURE";

export const GET_AGENT_DETAILS = "app/AddOrEditAgent/GET_AGENT_DETAILS";
export const GET_AGENT_DETAILS_SUCCESS = "app/AddOrEditAgent/GET_AGENT_DETAILS_SUCCESS";
export const GET_AGENT_DETAILS_FAILURE = "app/AddOrEditAgent/GET_AGENT_DETAILS_FAILURE";
