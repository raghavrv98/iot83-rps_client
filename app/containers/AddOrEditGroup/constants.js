/*
 *
 * AddOrEditGroup constants
 *
 */

export const ON_SUBMIT_REQUEST = "app/AddOrEditGroup/ON_SUBMIT_REQUEST";
export const ON_SUBMIT_REQUEST_SUCCESS = "app/AddOrEditGroup/ON_SUBMIT_REQUEST_SUCCESS";
export const ON_SUBMIT_REQUEST_FAILURE = "app/AddOrEditGroup/ON_SUBMIT_REQUEST_FAILURE";

export const GET_GROUP_DETAILS = "app/AddOrEditGroup/GET_GROUP_DETAILS";
export const GET_GROUP_DETAILS_SUCCESS = "app/AddOrEditGroup/GET_GROUP_DETAILS_SUCCESS";
export const GET_GROUP_DETAILS_FAILURE = "app/AddOrEditGroup/GET_GROUP_DETAILS_FAILURE";
