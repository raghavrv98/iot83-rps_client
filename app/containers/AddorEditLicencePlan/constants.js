/*
 *
 * AddorEditLicencePlan constants
 *
 */

export const DEFAULT_ACTION = "app/AddorEditLicencePlan/DEFAULT_ACTION";

export const ON_SUBMIT_REQUEST = "app/AddorEditLicencePlan/ON_SUBMIT_REQUEST";
export const ON_SUBMIT_REQUEST_SUCCESS = "app/AddorEditLicencePlan/ON_SUBMIT_REQUEST_SUCCESS";
export const ON_SUBMIT_REQUEST_FAILURE = "app/AddorEditLicencePlan/ON_SUBMIT_REQUEST_FAILURE";

export const GET_PLANS_LIST = "app/AddorEditLicencePlan/GET_PLANS_LIST";
export const GET_PLANS_LIST_SUCCESS = "app/AddorEditLicencePlan/GET_PLANS_LIST_SUCCESS";
export const GET_PLANS_LIST_FAILURE = "app/AddorEditLicencePlan/GET_PLANS_LIST_FAILURE";