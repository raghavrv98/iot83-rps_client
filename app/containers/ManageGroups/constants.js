/*
 *
 * ManageGroups constants
 *
 */

export const DEFAULT_ACTION = "app/ManageGroups/DEFAULT_ACTION";

export const GET_LIST = "app/ManageGroups/GET_LIST";
export const GOT_LIST = "app/ManageGroups/GOT_LIST";
export const NOT_GOT_LIST = "app/ManageGroups/NOT_GOT_LIST";


export const DELETE_REQUEST = "app/ManageGroups/DELETE_REQUEST";
export const DELETE_REQUEST_SUCCESS = "app/ManageGroups/DELETE_REQUEST_SUCCESS";
export const DELETE_REQUEST_FAILURE = "app/ManageGroups/DELETE_REQUEST_FAILURE";