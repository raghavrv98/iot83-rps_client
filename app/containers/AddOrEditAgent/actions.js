/*
 *
 * AddOrEditAgent actions
 *
 */

import { CREATE_AGENT_REQUEST,GET_AGENT_DETAILS } from "./constants";

export function createAgent(data) {
  return {
    type: CREATE_AGENT_REQUEST,
    payload: data,
  };
}
export function getAgentDetails(id) {
  return {
    type: GET_AGENT_DETAILS,
    id
  };
}
