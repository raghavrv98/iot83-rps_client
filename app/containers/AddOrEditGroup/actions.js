/*
 *
 * AddOrEditGroup actions
 *
 */

import { ON_SUBMIT_REQUEST,GET_GROUP_DETAILS } from "./constants";

export function onSubmitHandler(payload,id) {
  return {
    type: ON_SUBMIT_REQUEST,
    payload,
    id
  };
}

export function getGroupDetails(id) {
  return {
    type: GET_GROUP_DETAILS,
    id
  };
}
