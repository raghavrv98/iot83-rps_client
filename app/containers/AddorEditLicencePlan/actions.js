/*
 *
 * AddorEditLicencePlan actions
 *
 */

import { DEFAULT_ACTION, ON_SUBMIT_REQUEST, GET_PLANS_LIST } from "./constants";

export function defaultAction() {
  return {
    type: DEFAULT_ACTION
  };
}

export function onSubmitHandler(payload) {
  return {
    type: ON_SUBMIT_REQUEST,
    payload
  };
}

export function getPlansList(){
  return{
    type: GET_PLANS_LIST
  }
}