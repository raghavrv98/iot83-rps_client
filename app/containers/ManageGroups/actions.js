/*
 *
 * ManageGroups actions
 *
 */

import { DEFAULT_ACTION,GET_LIST ,DELETE_REQUEST} from "./constants";

export function defaultAction() {
  return {
    type: DEFAULT_ACTION
  };
}

export function getGroupList() {
  return {
    type: GET_LIST
  };
}

export function deleteHandler(id) {
  return {
    type: DELETE_REQUEST,
    id
  }
}