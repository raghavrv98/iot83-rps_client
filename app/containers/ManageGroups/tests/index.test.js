import React from 'react';
import { shallow } from 'enzyme';
import { ManageGroups, mapDispatchToProps, mapStateToProps } from '../index';
import { getGroupList, deleteHandler } from '../actions';
import { GET_LIST, DELETE_REQUEST } from "../constants";
import history from 'utils/history';
const mockGroupList = [
  {
    "id": "9f1c2f19a97d4a01bebec08ddd206fe2",
    "name": "DEFAULT",
    "description": "Default container",
    "mqttConnectionId": null,
    "devices": 0
  }]

describe('<ManageGroups />', () => {

  it('should test component when recieve props', () => {
    spyOn(ManageGroups.prototype, 'componentWillReceiveProps').and.callThrough();
    const wrapper = shallow(<ManageGroups getGroupList={getGroupList} list={undefined} listError={undefined} deleteSuccess={undefined} deleteFailure={undefined} />);
    expect(wrapper).toBeDefined();
    wrapper.setState({groupList: mockGroupList})
    wrapper.setProps({ list: mockGroupList, listError: "No Error Found.", deleteSuccess: "9f1c2f19a97d4a01bebec08ddd206fe2", deleteFailure: "No Error Found in Delete" });
    expect(ManageGroups.prototype.componentWillReceiveProps).toHaveBeenCalledTimes(1);
  })

  // it("clicking add button trigger onClick event", async () => {
  //   let expectedCount = history.length + 1;
  //   const wrapper = shallow(<ManageGroups getGroupList={getGroupList} history={history} />);
  //   let addButton = wrapper.find("button.btn.btn-create");
  //   expect(addButton.length).toBe(1);
  //   addButton.simulate('click');
  //   expect(history.length).toEqual(expectedCount);
  //   expect(history.location.pathname).toEqual("/addOrEditGroup");
  // });

  // it("clicking edit button trigger onClick event", async () => {
  //   const match = {
  //     params: {
  //         id: "demoId"
  //     }
  //   }
  //   let expectedCount = history.length + 1;
  //   const wrapper = shallow(<ManageGroups getGroupList={getGroupList} history={history} />);
  //   let addButton = wrapper.find("button.btn-transparent");
  //   expect(addButton.length).toBe(1);
  //   addButton.simulate('click');
  //   expect(history.length).toEqual(expectedCount);
  //   expect(history.location.pathname).toEqual("/addOrEditGroup/" + match.params.id);
  // });

  it('match snapshots', () => {
    const wrap = shallow(<ManageGroups getGroupList={getGroupList} />)
    expect(wrap).toMatchSnapshot()
  })

  it("test componentDidMount", async () => {
    const match = {
        params: {
            id: "demoId"
        }
    };
    const getGroupList = jest.fn();
    spyOn(ManageGroups.prototype, "componentDidMount").and.callThrough();
    shallow(
        <ManageGroups
            history={history}
            match={match}
            getGroupList={getGroupList}
        />
    );
    expect(ManageGroups.prototype.componentDidMount).toHaveBeenCalledTimes(1);
    expect(getGroupList).toHaveBeenCalledTimes(1);
  });

  it("Function testing on mapDispatchToProps", () => {
    const payload = "demo";
    const mockParams = {
      params: {
        id: "demo"
      }
    };
    const dispatch = jest.fn();
    mapDispatchToProps(dispatch).getGroupList();
    expect(dispatch.mock.calls[0][0]).toEqual({
      type: GET_LIST
    });
    mapDispatchToProps(dispatch).deleteHandler(mockParams.params.id);
    expect(dispatch.mock.calls[1][0]).toEqual({
      type: DELETE_REQUEST,
      id: "demo"
    });
  });
});
