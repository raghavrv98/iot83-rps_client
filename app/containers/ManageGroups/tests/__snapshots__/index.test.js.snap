// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ManageGroups /> match snapshots 1`] = `
<div
  className="appContent"
>
  <HelmetWrapper
    defer={true}
    encodeSpecialCharacters={true}
  >
    <title>
      ManageGroups
    </title>
    <meta
      content="Description of ManageGroups"
      name="description"
    />
  </HelmetWrapper>
  <div
    className="pageBreadcrumb"
  >
    <div
      className="flex-item fx-b70"
    >
      <p>
        <FormattedMessage
          defaultMessage="Identity Access Management"
          id="app.common.messages.iam"
          values={Object {}}
        >
          <Component />
        </FormattedMessage>
      </p>
      <h5>
        <FormattedMessage
          defaultMessage="All Available Groups"
          id="app.containers.ManageGroup.title"
          values={Object {}}
        >
          <Component />
        </FormattedMessage>
      </h5>
    </div>
    <div
      className="flex-item fx-b30 text-right align-items-center"
    />
  </div>
  <SkeletonLoader
    mode="fullView"
    skeleton="skeletonReactTable"
  />
</div>
`;
